Feature ab 46 admin the admins

BerendWouters opened the pull request:
This PR introduces the Admin management module.
To make sure you can run this branch, you will need to delete the following entities in the database. The Seed script will take care of it all:
Deletable entities:
- APIResources
- Clients

BerendWouters committed:
feat(admin) add admin table

AB#46

BerendWouters committed:
feat(admin) show details of admin user

AB#46

BerendWouters committed:
merge with develop

BerendWouters committed:
feat(admin) persists adminUser in the database

AB#46

BerendWouters committed:
feat(admin) create and update admin users

AB#46

BerendWouters committed:
feat(user) create users with temp password for login

AB#46

BWouters committed:
feat(admin) adds claim based user authorization

AB#46

BWouters committed:
feat(admin) introduce isAdmin information in authService

AB#46

vyruz1986 commented:
>To make sure you can run this branch, you will need to delete the following entities in the database. The Seed script will take care of it all:

So do we have to do this manually, or will the seed script take care of it? I'm confused...
More importantly, if we have to do it manually, how will we do this on Stijn's environment?

vyruz1986 reviewed (changes requested):
Several remarks which need to be addressed:
* latest develop is not merged in this branch, trying to do so results in merge conflicts
* debugging this locally doesn't work for me, I don't get the 'beheerders' button in the menu, neither when removing all records from the `ApiResources` and `Clients` tables

BerendWouters commented:
> > To make sure you can run this branch, you will need to delete the following entities in the database. The Seed script will take care of it all:
> 
> So do we have to do this manually, or will the seed script take care of it? I'm confused...
> More importantly, if we have to do it manually, how will we do this on Stijn's environment?

Will fix this by creating a new migration dropping all users.

BWouters committed:
merge with develpo

BWouters committed:
merge fixes

BWouters committed:
feat(admin) adds Sql script to cleanup database regarding admins

AB#46

BerendWouters commented:
> Several remarks which need to be addressed:
> 
> * latest develop is not merged in this branch, trying to do so results in merge conflicts
> * debugging this locally doesn't work for me, I don't get the 'beheerders' button in the menu, neither when removing all records from the `ApiResources` and `Clients` tables

Remarks have been taken into account:

- [x] Develop is merged into branch. Conflicts solved.
- [x] Debugging should be possible, admin is now updated with correct claims
* Additionally I've added a script allowing you to change any other user to admin. Note, this script also deletes Clients and APIResources

vyruz1986 commented on Singer.API/Services/UserService.cs:
```diff
@@ -40,7 +40,7 @@ protected UserService(ApplicationDbContext context, IMapper mapper, UserManager<
          };
 
-         var userCreationResult = await UserManager.CreateAsync(baseUser);
+         var userCreationResult = await UserManager.CreateAsync(baseUser, "Testpassword123!");
```
Please add `//TODO` that this should be replaced with random password once we have an email service in place which can send out random passwords.
Alternatively, you could go the extra mile and add a random generated password now already, which is logged to the console or something so it can be retrieved from the logs.

BWouters committed:
doc adds ToDo
